web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -14,18 +19,112 @@ interface FlagsProps {
   data?: FlagMap;
 }
 
+const compareAlphaFn = (keys: boolean, reverse: boolean) => (
+  [k1, v1]: [string, string],
+  [k2, v2]: [string, string]
+): number => {
+  const a = keys ? k1 : v1;
+  const b = keys ? k2 : v2;
+  const reverser = reverse ? -1 : 1;
+  return reverser * a.localeCompare(b);
+};
+
+const boldSubstring = (s: string, subs: string) => {
+  const matchResult = fuzzy.match(subs, s, {
+    pre: '<strong>',
+    post: '</strong>',
+  });
+  if (matchResult == null) {
+    return s;
+  }
+  return matchResult.rendered;
+};
+
+const getSortIcon = (b: boolean | undefined): IconDefinition => {
+  if (b === undefined) {
+    return faSort;
+  }
+  if (b) {
+    return faSortDown;
+  }
+  return faSortUp;
+};
+
+interface SortState {
+  name: string;
+  alpha: boolean;
+  focused: boolean;
+}
+
 export const FlagsContent: FC<FlagsProps> = ({ data = {} }) => {
+  const initialSearch = '';
+  const [searchState, setSearchState] = useState(initialSearch);
+  const initialSort: SortState = {
+    name: 'Flag',
+    alpha: true,
+    focused: true,
+  };
+  const [sortState, setSortState] = useState(initialSort);
   return (
     <>
       <h2>Command-Line Flags</h2>
-      <Table bordered size="sm" striped>
+      <InputGroup>
+        <Input
+          autoFocus
+          placeholder="Filter by flag name or value..."
+          className="my-3"
+          value={searchState}
+          onChange={({ target }: ChangeEvent<HTMLInputElement>): void => {
+            setSearchState(target.value);
+          }}
+        />
+      </InputGroup>
+      <Table bordered size="sm" striped hover>
+        <thead>
+          <tr>
+            {['Flag', 'Value'].map((col: string) => (
+              <td
+                key={col}
+                className={`px-4 ${col}`}
+                style={{ width: '50%' }}
+                onClick={(): void =>
+                  setSortState({
+                    name: col,
+                    focused: true,
+                    alpha: sortState.name === col ? !sortState.alpha : true,
+                  })
+                }
+              >
+                <span className="mr-2">{col}</span>
+                <FontAwesomeIcon icon={getSortIcon(sortState.name !== col ? undefined : sortState.alpha)} />
+              </td>
+            ))}
+          </tr>
+        </thead>
         <tbody>
-          {Object.keys(data).map(key => (
-            <tr key={key}>
-              <th>{key}</th>
-              <td>{data[key]}</td>
-            </tr>
-          ))}
+          {Object.entries(data)
+            .sort(compareAlphaFn(sortState.name === 'Flag', !sortState.alpha))
+            .map(([flag, value]) => [`--${flag}`, value])
+            .filter(([flag, value]) => flag.includes(searchState) || value.includes(searchState))
```

Comment:
Hmm, now we are only using the fuzzy lib for highlighting, not for the filtering. But I can see how that could get a bit complex because you have both the flag name and value, and if either of them matches, you want to display both...

Reply:
Then again, if the filtering is stricter than the highlighting of how the input text applies to the search items, I think this will be confusing... maybe we could use `fuzzy.match()` here too (checking for `null`), just to check if an item would be returned by the fuzzy lib, vs. `flag.includes()`?

Reply:
Good point. In the commit below I used fuzzy.filter on `--${flag} ${value}` (joined by space) and then split it (by space) when rendering into table cells. Thoughts on that approach?